doc/SAI-Proposal-PoE.md

```diff
@@ -0,0 +1,591 @@
+
+# Overview
+
+Power over Ethernet, or PoE, describes any of several standards that pass electric power along with data on twisted-pair Ethernet cabling. This allows a single cable to provide both a data connection and enough electricity to power networked devices such as wireless access points (WAPs), IP cameras and VoIP phones.
+
+# Proposal
+
+The current SAI does not support PoE. We will implement PoE in a way that any vendor can provide their own PoE hardware and software solution.
+
+## Abbreviations and Information
+
+| **Term** | **Definition**                                                                              |
+| -------------- | ------------------------------------------------------------------------------------------------- |
+| PoE            | Power over Ether                                                                                  |
+| PSE            | Power Sourcing Equipment                                                                          |
+| PD             | Powered Device, can power the device through an PD-capable Ethernet port                          |
+| PoE Port       | Ethernet port that can deliver power and info to devices, located on the front panel of the board |
+
+# Key Ideas behind PoE
+
+1. PoE provides a hardware solution for providing a data connection and electricity to power connected devices
+2. The PoE software delivers information about the board and its operational statuses
+3. Users can see PoE information and configure PoE features according to their needs
+
+# PoE Hardware Design Example
+
+![poe-physical-design](figures/PoE_Physical_Design.png)
+
+# SAI POE Components
+
+## New PoE SAI objects
+
+Three new SAI objects are introduced.
+
+### PoE Device ID
+
+This object is used to hold the global configuration and PoE hardware status. In many cases, one PoE device instance will have access to all the PoE hardware and components.
+
+### PoE PSE ID
+
+This object is used to access data from the Power Sourcing Equipment devices in the board. Such as PSE temperatures, versions, and statuses. There can be multiple PSEs in one board.
+
+### PoE Port ID
+
+This object is used to access data on a port level. PoE ports supply power using an ethernet cable. PoE port object gives access to read data and configure these ports.
+
+## New SAI Switch Type PoE
+
+A new abstraction type for PoE is created which allows multiple vendors to implement their own PoE software and hardware solutions.
+
+```
+/**
+ * @brief Attribute data for #SAI_SWITCH_ATTR_TYPE
+ */
+typedef enum _sai_switch_type_t
+{
+    /** Switch type is Switching Network processing unit */
+    SAI_SWITCH_TYPE_NPU,
+
+    /** Switch type is PHY */
+    SAI_SWITCH_TYPE_PHY,
+
+    /** Switch type is VOQ based NPU */
+    SAI_SWITCH_TYPE_VOQ,
+
+    /** Switch type is Fabric switch device */
+    SAI_SWITCH_TYPE_FABRIC,
+
+    /** Switch type is PoE (Power over Ethernet) */
+    SAI_SWITCH_TYPE_POE,
```

Comment:
why is this a new type of switch?

Reply:
what is the purpose of this "SAI_SWITCH_TYPE_POE" ? Does this mean the switch is PoE capable and can deliver power via it's port?

Reply:
Yes.

If the switch and PoE vendors differ than we want to provide flexibility to create two instances for each functionality. (Different hardware, two different vendors, two different implementations)
For example, if vendor A already has a SAI implementation and they want external PoE support, they can keep using their solution and add another PoE implementation from vendor B.

(If the same vendor provides a NPU and PoE switch, then it can be placed under NPU switch with PoE abilities.
Per the community's request, I added a new PoE attribute to sai port: SAI_PORT_ATTR_POE_PORT_ID)

Reply:
Ok.